docs/proposal/20250324-rosa-iam-roles.md

```diff
@@ -0,0 +1,226 @@
+              prefixPolicy: ""
+            groups:
+              claim: groups
+          oidcClients:
+            - componentName: console
+              componentNamespace: openshift-console
+              clientID: <console-client-id>
+              clientSecret:
+                name: console-client-secret
+    status:
+      conditions:
+        - type: Ready
+          status: "True"
+          message: "ROSA HCP Roles & OIDC created"
+          reason: "Created"
+      # Created oidc config and provider.      
+      oidcID: "123456789123456789abcdefgh"
+      oidcProviderARN: arn:aws:iam::12345678910:oidc-provider/oidc.ocp.dev.org/123456789123456789abcdefgh
+
+      # Created ROSA-HCP managed account roles
+      accountRolesRef:
+        installerRoleARN: "arn:aws:iam::12345678910:role/capa-HCP-ROSA-Installer-Role"
+        supportRoleARN: "arn:aws:iam::12345678910:role/capa-HCP-ROSA-Support-Role"
+        workerRoleARN: "arn:aws:iam::12345678910:role/capa-HCP-ROSA-Worker-Role"      
+
+      # Created ROSA-HCP managed operator roles
+      operatorRolesRef:
+        ingressARN: "arn:aws:iam::12345678910:role/capa-openshift-ingress-operator-cloud-credentials"
+        imageRegistryARN: "arn:aws:iam::12345678910:role/capa-openshift-image-registry-installer-cloud-credentials"
+        storageARN: "arn:aws:iam::12345678910:role/capa-openshift-cluster-csi-drivers-ebs-cloud-credentials"
+        networkARN: "arn:aws:iam::12345678910:role/capa-openshift-cloud-network-config-controller-cloud-credent"
+        kubeCloudControllerARN: "arn:aws:iam::12345678910:role/capa-kube-system-kube-controller-manager"
+        nodePoolManagementARN: "arn:aws:iam::12345678910:role/capa-kube-system-capa-controller-manager"
+        controlPlaneOperatorARN: "arn:aws:iam::12345678910:role/capa-kube-system-control-plane-operator"
+        kmsProviderARN: "arn:aws:iam::12345678910:role/capa-kube-system-kms-provider"      
+    ```
+
+    The ROSARoleConfig status conditions should present the roles creation order and failed status. Below are examples of condition state.
+
+    ```yaml
+    status:
+      conditions:
+        - type: Ready
+          status: "False"
+          message: "Creating OIDC provider"
+          reason: "Creating"
+    ```
+
+    ```yaml
+    status:
+      conditions:
+        - type: Ready
+          status: "True"
+          message: "ROSA HCP Roles & OIDC created"
+          reason: "Created"
+    ```
+
+    ```yaml
+    status:
+      conditions:
+        - type: Ready
+          status: "False"
+          message: "Failed to create IAM operator roles oidc-config not exist."
+          reason: "Failed"
+    ```
+
+2. **Implement a CAPA controller leveraging the ROSA library**
+   - Manage CRUD operations for ROSA HCP IAM roles and OIDC configurations.
+   - Ensure proper IAM permissions and security best practices.
+
+3. **Validate and test integration**
+   - Develop unit tests for ROSA HCP IAM and OIDC CRD reconciliation.
```

Comment:
We need to re-open discussions about getting some community-owned, ROSA-backed clusters for CI. Right now, I think most of this testing would happen outside of this repo.

Reply:
I agree, we need to have CI test AND yes,  there are some test done out of the CAPA repo.